### ISSUE TITLE
Topic modeling_pyLDAvis_Blog

### ISSUE THREAD

**Issue by Nohyunsun** [Nohyunsun/ADHD-Self-Diagnosis-Across-Platforms#14]
[Topicmodeling_blog_pyLDAvis.ipynb](https://github.com/user-attachments/files/23737663/Topicmodeling_blog_pyLDAvis.ipynb)

**Comment by Nohyunsun** [comment]
# ✍️ Blog Topic Modeling — ADHD Self-Diagnosis  
LDA • CountVectorizer • Topic Keyword Extraction • pyLDAvis Topic Map

This repository contains a complete topic modeling pipeline for **Naver Blog, Tistory, and other long-form blog content** related to ADHD self-diagnosis.  
Blog posts are typically **long, detailed, reflective, and information-archival**, making them ideal for in-depth topic modeling.

This notebook includes:

- Preprocessing & tokenization for long-form text  
- CountVectorizer-based text representation  
- LDA topic modeling  
- Topic keyword ranking  
- Topic map visualization (pyLDAvis)  
- Interpretation table for discovered topics  

---

## 🚀 1. Overview

Blog content differs significantly from Instagram·X:

- Longer, paragraph-level narrative  
- Rich descriptions (diagnosis process, treatment history, daily experiences)  
- High information density  
- Often SEO-optimized, producing clear keyword clusters  

This notebook identifies latent topics around:

- ADHD symptom descriptions  
- Diagnosis experiences  
- Medication and treatment stories  
- Self-help and coping strategies  
- Parenting narratives (if included)  

---

## 📦 2. Repository Structure
adhd-blog-topicmodeling/
│
├── Topicmodeling_blog_pyLDAvis.ipynb # Main notebook
│
├── data/
│ └── blog_cleaned_text.csv # Preprocessed text (not included)
│
├── output/
│ ├── lda_topics_blog.csv
│ ├── lda_keywords_blog.csv
│ ├── topic_map_blog.html # pyLDAvis visualization
│ ├── topic_summary_blog.csv
└── README.md

---

## 🧹 3. Data Requirements

Expected dataset structure:

| Column | Description |
|--------|-------------|
| `text` | Blog post body text |
| `created_at` | (optional) post date |
| `platform` | optional |

Preprocessing pipeline accounts for:

- Removing HTML tags (common in crawled blog data)  
- Stripping URLs  
- Removing emojis & special characters  
- Normalizing spacing  
- Tokenization via Okt/Mecab  
- Stopword cleaning  

Example implementation:

```python
def clean_text(t):
    t = re.sub(r"<.*?>", " ", t)      # remove HTML
    t = re.sub(r"http\S+|www\S+", "", t)
    t = re.sub(r"[^가-힣A-Za-z0-9\s]", " ", t)
    return t.strip()

**Comment by Nohyunsun** [comment]
## 🧠 6. LDA Topic Modeling
```python
lda = LatentDirichletAllocation(
    n_components=K,        # e.g., 8–15 for long-form text
    learning_method='batch',
    random_state=42
)
lda.fit(text_vec)
```
Extracting representative keywords :
```python
topic_keywords = get_top_words(lda, vectorizer.get_feature_names_out(), 20)
```

**Comment by Nohyunsun** [comment]
## 🗺 7. pyLDAvis Topic Map
```python
vis = pyLDAvis.sklearn.prepare(lda, text_vec, vectorizer)
pyLDAvis.save_html(vis, "topic_map_blog.html")
```
🔹 Lambda (λ) Tuning
- λ = 0 → frequency-based
- λ = 1 → topic-specific terms
- For blogs, λ = 0.8 tends to produce the best topic clarity,
  because documents are long enough to form distinct semantic clusters.

**Comment by Nohyunsun** [comment]
## 📊 8. Topic Interpretation Table
Example structure : 

<img width="1268" height="423" alt="Image" src="https://github.com/user-attachments/assets/6c9de605-5a2e-4a41-b2a6-51110fccd4d7" />
(Actual topics match notebook output)

**Comment by Nohyunsun** [comment]
## 📈 9. Optional: Yearly Topic Trends
```python
df['year'] = pd.to_datetime(df['created_at']).dt.year
df.groupby(['year', 'topic']).size()
```
Useful for :
- ADHD awareness growth across years
- Changes in blog content SEO trends
- Rise of self-diagnosis discussions
